wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -643,6 +643,15 @@ impl PhysicalDeviceCapabilities {
             }
         }
 
+        if self.effective_api_version >= vk::API_VERSION_1_2
+            || self.supports_extension(vk::KhrImageFormatListFn::name())
+        {
+            // Optional `VK_KHR_swapchain_mutable_format`
+            if self.supports_extension(vk::KhrSwapchainMutableFormatFn::name()) {
+                extensions.push(vk::KhrSwapchainMutableFormatFn::name());
+            }
+        }
```

Comment:
```suggestion
        // Optional `VK_KHR_swapchain_mutable_format`
        if self.supports_extension(vk::KhrSwapchainMutableFormatFn::name()) {
            extensions.push(vk::KhrSwapchainMutableFormatFn::name());
        }
```

I'm quite certain that `supports_extension` returns `false` if the required extensions for this one are not supported.